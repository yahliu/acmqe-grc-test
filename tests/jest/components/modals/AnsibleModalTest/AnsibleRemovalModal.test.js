/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import { renderAnsibleRemovalModal } from '../../../../../src-web/components/modals/AnsibleAutomationModal/AnsibleRemovalModal'

describe('render ansible removal modal', () => {
  it('render opened ansible removal modal', () => {
    const component = renderAnsibleRemovalModal({
      openDelModal:true,
      policyAutoName: 'test-policy-3-1624936287-policy-automation', locale: 'us',
      handleDeleteClick:jest.fn(), handleCloseDelModal:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render opened ansible removal modal without policy automation name', () => {
    const component = renderAnsibleRemovalModal({
      openDelModal:true,
      policyAutoName: '', locale: 'us',
      handleDeleteClick:jest.fn(), handleCloseDelModal:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render closed ansible removal modal', () => {
    const component = renderAnsibleRemovalModal({
      openDelModal:false,
      policyAutoName: 'test-policy-3-1624936287-policy-automation', locale: 'us',
      handleDeleteClick:jest.fn(), handleCloseDelModal:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render closed ansible removal modal without policy automation name', () => {
    const component = renderAnsibleRemovalModal({
      openDelModal:true,
      policyAutoName: '', locale: 'us',
      handleDeleteClick:jest.fn(), handleCloseDelModal:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })
})
