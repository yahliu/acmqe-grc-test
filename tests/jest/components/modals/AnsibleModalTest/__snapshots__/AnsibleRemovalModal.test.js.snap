// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ansible removal modal render closed ansible removal modal 1`] = `
<AcmModal
  actions={
    Array [
      <AcmButton
        onClick={[MockFunction]}
        variant="danger"
      >
        Delete automation
      </AcmButton>,
      <AcmButton
        onClick={[MockFunction]}
        variant="link"
      >
        Cancel
      </AcmButton>,
    ]
  }
  id="remove-ansible-modal"
  isOpen={false}
  onClose={[MockFunction]}
  showClose={true}
  title="Permanently delete policy violation automation?"
  titleIconVariant="warning"
  variant="medium"
>
  If you delete {0}, you will permanently lose all of its information.
</AcmModal>
`;

exports[`render ansible removal modal render closed ansible removal modal without policy automation name 1`] = `
<AcmModal
  actions={
    Array [
      <AcmButton
        onClick={[MockFunction]}
        variant="danger"
      >
        Delete automation
      </AcmButton>,
      <AcmButton
        onClick={[MockFunction]}
        variant="link"
      >
        Cancel
      </AcmButton>,
    ]
  }
  id="remove-ansible-modal"
  isOpen={true}
  onClose={[MockFunction]}
  showClose={true}
  title="Permanently delete policy violation automation?"
  titleIconVariant="warning"
  variant="medium"
>
  If you delete {0}, you will permanently lose all of its information.
</AcmModal>
`;

exports[`render ansible removal modal render opened ansible removal modal 1`] = `
<AcmModal
  actions={
    Array [
      <AcmButton
        onClick={[MockFunction]}
        variant="danger"
      >
        Delete automation
      </AcmButton>,
      <AcmButton
        onClick={[MockFunction]}
        variant="link"
      >
        Cancel
      </AcmButton>,
    ]
  }
  id="remove-ansible-modal"
  isOpen={true}
  onClose={[MockFunction]}
  showClose={true}
  title="Permanently delete policy violation automation?"
  titleIconVariant="warning"
  variant="medium"
>
  If you delete {0}, you will permanently lose all of its information.
</AcmModal>
`;

exports[`render ansible removal modal render opened ansible removal modal without policy automation name 1`] = `
<AcmModal
  actions={
    Array [
      <AcmButton
        onClick={[MockFunction]}
        variant="danger"
      >
        Delete automation
      </AcmButton>,
      <AcmButton
        onClick={[MockFunction]}
        variant="link"
      >
        Cancel
      </AcmButton>,
    ]
  }
  id="remove-ansible-modal"
  isOpen={true}
  onClose={[MockFunction]}
  showClose={true}
  title="Permanently delete policy violation automation?"
  titleIconVariant="warning"
  variant="medium"
>
  If you delete {0}, you will permanently lose all of its information.
</AcmModal>
`;
